### PR TITLE
Issue/13326 fix threat from threat details

### PR DESCRIPTION
**Pull request by ashiagr**
Parent #13326

This PR
- Reuses `FixThreatsUsecase` for fix threat action on the threat details screen.
- When fix request succeeds, navigates back to the scan state screen and fetches fix status by reusing existing logic for fix threats status.
- Moves ignore threat success message to the scan state screen as per https://github.com/wordpress-mobile/WordPress-Android/pull/13768#discussion_r559502715.

https://user-images.githubusercontent.com/1405144/105171373-26b17580-5b44-11eb-9374-8db7ec513e17.mov

https://user-images.githubusercontent.com/1405144/105170568-1e0c6f80-5b43-11eb-9994-42c1744acb33.mov

To test:

Prerequisite:

- Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
- You have access to WP.com account with a site having scan capability and threats (e.g. https://pressable-jetpack-daily-scan.mystagingwebsite.com). Server credentials should be set on the site.
- A threat is added to the site. In case you run out of threats, re-install bad calendar plugin from [here](https://jetpackp2.files.wordpress.com/2020/05/calendar-1.zip) and re-run scan.

1. Login to the app using above WP.com account and select the site on the My Site tab
2. Click scan menu item on My Site
3. Click a threat on scan state screen
4. Click fix threat button on the threat details screen
5. Notice that a confirmation dialog is shown (see video)
6. Click ok in the confirmation dialog
7. Notice that app navigates back to scan state screen and fix status is shown

Repeat above steps and enable airplane mode after step 6. Notice that an error message is displayed and actions buttons are re-enabled.

Also test that when ignore threat action succeeds, ignore threat success message is shown on the scan state screen. 

(Tests included in `ThreatDetailsViewModelTest` , `ScanViewModelTest`)

Notes:
1. Ignore all styling
2. Server credentials check is not handled
3. Fix status fine-tuning will be done separately
4. Task for pluralisation of messages (threat vs threats based on threat count) is added to backlog

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt [hunk 1]
```diff
@@ -125,10 +125,16 @@ class ScanFragment : Fragment(R.layout.scan_fragment) {
...
-        ) {
-            viewModel.onScanStateRequested()
+        if (requestCode == RequestCodes.SHOW_THREAT_DETAILS) {
+            data?.let {
+                val threatId = it.getLongExtra(ThreatDetailsFragment.REQUEST_FIX_STATE, 0L)
+                val messageRes = it.getIntExtra(ThreatDetailsFragment.REQUEST_SCAN_STATE, 0)
+                if (threatId > 0L) {
+                    viewModel.onFixStateRequested(threatId)
+                } else {
+                    viewModel.onScanStateRequestedWithMessage(messageRes)
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt:135]: ⚠️ The `else` will be performed even when `data?.getBooleanExtra(ThreatDetailsFragment.REQUEST_SCAN_STATE, false) == false` (messageRes == 0) -> is this change intended?

**Review by malinajirka — APPROVED** [review]
Thanks @ashiagr! It looks good to me. Feel free to merge it after you consider the comment.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt:135]: No that's not intended. Thanks for catching it 🙇‍♀️. 
Added if check in 562ea14

**Comment by ashiagr** [comment]
Thanks for the review @malinajirka! I've addressed the review comment.
Ready for another look.
